### ISSUE TITLE
Pubsub across a broker cluster not working

### ISSUE THREAD

**Issue by card4ash** [erlio/vernemq#586]
### Environment

- VerneMQ Version: vernemq_1.2.2-1_amd64
- OS:
Ubuntu 16.04 LTS
- Erlang/OTP version (if building from source):

### Expected behavior
Node A and B are in a cluster.
Client A subscribes to the topic "/test" on a node A.
Client B publishes a message to the topic "/test" on a node B.
Client A receives the message.

### Actual behaviour

-Client A doesn't receive a message.

cluster form
+----------------------+-------+
|         Node         |Running|
+----------------------+-------+
|VerneMQ@192.168.11.218| true  |
|VerneMQ@192.168.11.92 | true  |
+----------------------+-------+

**Comment by ioolkos** [comment]
Hi @card4ash 
`listener.vmq.clustering = 127.0.0.1:44053` in the vernemq.conf.
Can your nodes communicate (ping each other) on that port?

**Comment by card4ash** [comment]
I have tried with listener.vmq.clustering = 127.0.0.1:44053 but no luck
also tried with listener.vmq.clustering = 192.168.11.92:44053 and 192.168.11.218:44053
and also with 0.0.0.0:44053 on both nodes but no luck in any case. no firewall involved. 
netstat show this port listen.

**Comment by card4ash** [comment]
`allow_anonymous = on
allow_register_during_netsplit = on
allow_publish_during_netsplit = on
allow_subscribe_during_netsplit = on
allow_unsubscribe_during_netsplit = on
allow_multiple_sessions = on
max_inflight_messages = 20
max_online_messages = 1000
max_offline_messages = 1000
max_message_size = 0
upgrade_outgoing_qos = off
listener.max_connections = 10000
listener.nr_of_acceptors = 10
listener.tcp.default = 192.168.11.92:1883
listener.vmq.clustering = 0.0.0.0:44053
listener.http.default = 192.168.11.92:8888
listener.mountpoint = off
systree_enabled = on
systree_interval = 20000
graphite_enabled = off
graphite_host = localhost
graphite_port = 2003
graphite_interval = 20000
shared_subscription_policy = prefer_local
plugins.vmq_passwd = on
plugins.vmq_acl = on
plugins.vmq_diversity = off
plugins.vmq_webhooks = off
plugins.vmq_bridge = off
vmq_acl.acl_file = /etc/vernemq/vmq.acl
vmq_acl.acl_reload_interval = 10
vmq_passwd.password_file = /etc/vernemq/vmq.passwd
vmq_passwd.password_reload_interval = 10
vmq_diversity.script_dir = /usr/share/vernemq/lua
vmq_diversity.auth_postgres.enabled = off
vmq_diversity.auth_mysql.enabled = off
vmq_diversity.auth_mongodb.enabled = off
vmq_diversity.auth_redis.enabled = off
log.console = file
log.console.level = info
log.console.file = /var/log/vernemq/console.log
log.error.file = /var/log/vernemq/error.log
log.syslog = off
log.crash = on
log.crash.file = /var/log/vernemq/crash.log
log.crash.maximum_message_size = 64KB
log.crash.size = 10MB
log.crash.rotation = $D0
log.crash.rotation.keep = 5
nodename = VerneMQ@192.168.11.92
distributed_cookie = vmq
erlang.async_threads = 64
erlang.max_ports = 262144
leveldb.maximum_memory.percent = 70`

**Comment by larshesel** [comment]
Hi - do you see any errors in the logs? If the different nodes can't setup the data communications connection then it'll be shown in the logs with a message like `can't connect to cluster node <Node> due to <Reason>`.

Note, until VerneMQ 1.2.3 the `vmq-admin cluster show` would report the nodes reachable even if the data communications channel was down. See bullet point number two in the changelog for VerneMQ 1.2.3: https://github.com/erlio/vernemq/blob/master/changelog.md#vernemq-123

I hope this helps.

**Comment by card4ash** [comment]
```
2018-01-18 23:54:28.260 [info] <0.339.0>@vmq_cluster_node:connect:207 successfully connected to cluster node 'VerneMQ@192.168.11.218'
2018-01-18 23:54:28.260 [warning] <0.32715.2>@vmq_cluster_com:loop:74 terminate due to remote_node_not_available
```


```
2018-01-19 17:02:52.355 [info] <0.344.0>@vmq_cluster_node:connect:207 successfully connected to cluster node 'VerneMQ@192.168.11.92'
2018-01-19 17:02:52.356 [warning] <0.21298.8>@vmq_cluster_com:loop:74 terminate due to remote_node_not_available
2018-01-19 17:02:53.367 [info] <0.344.0>@vmq_cluster_node:connect:207 successfully connected to cluster node 'VerneMQ@192.168.11.92'
2018-01-19 17:02:53.367 [warning] <0.21299.8>@vmq_cluster_com:loop:74 terminate due to remote_node_not_available
```

**Comment by larshesel** [comment]
Those log messages indicate that the you likely have some kind of asymmetric cluster setup. This could happen if a node was administratively removed from the cluster while it was offline and in the comes back. Then it still believes it can connect to the other cluster nodes, but the other cluster nodes no longer know about it.

As it's hard to know what happened before and led into this situation, I would recommend wiping all cluster metadata (if this is viable option for you) by shutting down all the nodes and then remove the metadata (probably in `/var/lib/vernemq/meta/*`). This will mean each node no longer has any knowledge of the cluster. Then restart each node and re-create the cluster using the `vmq-admin cluster join` command.

Hope this helps.

**Comment by card4ash** [comment]
Thank you. This is helpful. Finally set up cluster successfully.

**Comment by nik736** [comment]
Sorry that I have to reopen this.

I've got the same error, Ubuntu 16.04 as well, using the provided 1.3.0 bin.

I am trying to setup a cluster but I always get
`@vmq_cluster_com:loop:74 terminate due to remote_node_not_available`

I basically have similar settings like card4ash but they don't see each other "running". Not sure what could cause this, distributed_cookie is the same on both servers.

I also tried stopping vernemq deleting the metadata (/var/lib/vernemq/meta/*) and starting verne again. Same issue.

Any idea what I could try to get it running?


Thank you.
